main.js

```diff
@@ -291,3 +291,32 @@ form.addEventListener('submit', formSubmit);
 emailInput.addEventListener('focus', () => {
   errorMessage.style.display = 'none';
 });
+
+/* Local storage */
+
+const userName = document.querySelector('.text-input');
+const messageInput = document.querySelector('textarea');
+
+userName.addEventListener('input', () => {
+  localStorage.setItem('userName', userName.value);
+});
+
+emailInput.addEventListener('input', () => {
+  localStorage.setItem('emailInput', emailInput.value);
+});
+
+messageInput.addEventListener('input', () => {
+  localStorage.setItem('messageInput', messageInput.value);
+});
+
+if (userName != null) {
+  userName.value = window.localStorage.userName;
+}
+
+if (emailInput != null) {
+  emailInput.value = window.localStorage.emailInput;
+}
+
+if (messageInput != null) {
+  messageInput.value = window.localStorage.messageInput;
+}
```

Comment:
- [ ] Kindly notice that the form data for all inputs should be stored in a single object, saved to a single local storage registry.

```
const formData = {
    name: nameInput.value,
    email: emailInput.value,
    messageInput: messageInput.value,
};
```

- [ ] Also in order to be saved to the local storage, the data object is turned into a string using `JSON.stringify` or similar methods. Ie:

```
localStorage.setItem('contactFormInput', JSON.stringify(formData));
```

- [ ] Also in order to be used after being retrieved from the local storage, the saved string is turned back into an object using `JSON.parse` or similar methods.

```
let savedFormData = localStorage.getItem('contactFormInput');
savedFormData = JSON.parse(savedFormData);
```

Reply:
**Thank you very much for the full feedback and examples!** @ShahierNashaat We'll work on those steps :raised_hands: